### ISSUE TITLE
False Negative JavadocType Unused param

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#17536]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadoctype.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
$ cat Test.java
public class Test {

    /**
     *
     * @param num1 a number
     * @param num2 a number
     * @param num3 a number
     * @param num4 a number
     * @return sum
     */
    public int method1(int num1, int num2) {
        int sum = num1 + num2;
        return sum;
    }

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="authorFormat" value="\S"/>
            <property name="allowMissingParamTags" value="true"/>
            <property name="allowUnknownTags" value="true"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-10.26.1-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
Expected
```
Starting audit...
[ERROR] Test.java:7:8: Unused @param tag for 'num3'. [JavadocType]
[ERROR] Test.java:8:8: Unused @param tag for 'num4'. [JavadocType]
Audit done.

```

**Comment by romani** [comment]
this Check is very old and doing too much, maintaining reliable logic is hard now in it.

We highly recommend to create new Check if any gaps are noticed in old Checks.

from description of Check:
> Checks the Javadoc comments for type definitions. 

so it does nothing for methods. you propbably mean to use https://checkstyle.org/checks/javadoc/javadocmethod.html

**Comment by Atharv3221** [comment]
Okay
